File: droneDSL/postprocess/python/interfaces/CloudletItf.py
# SPDX-FileCopyrightText: 2023 Carnegie Mellon University - Satyalab
#
# SPDX-License-Identifier: GPL-2.0-only

from abc import ABC, abstractmethod

class CloudletItf(ABC):

    @abstractmethod
    def processResults(self, result_wrapper):
        pass

    @abstractmethod
    def startStreaming(self, drone, model, sample_rate):
        pass

    @abstractmethod
    def stopStreaming(self):
        pass

    @abstractmethod
    def switchModel(self, model):
        pass

    @abstractmethod
    def sendFrame(self, frame):
        pass

    @abstractmethod
    def getResults(self, engine_key):
        pass
    
    @abstractmethod
    def clearResults(self, engine_key):
        pass

File: onboard/java/app/src/main/java/edu/cmu/cs/dronebrain/interfaces/CloudletItf.java
// SPDX-FileCopyrightText: 2023 Carnegie Mellon University - Satyalab
//
// SPDX-License-Identifier: GPL-2.0-only

package edu.cmu.cs.dronebrain.interfaces;
import org.json.JSONArray;

import java.util.Vector;

public interface CloudletItf {
    String ip = "127.0.0.1";
    void processResults(Object resultWrapper);
    void startStreaming(DroneItf drone, String model, Integer sample_rate);
    void switchModel(String model);
    void stopStreaming();
    void sendFrame(byte[] frame);
    JSONArray getResults(String key);
}
